setup.py

```diff
@@ -130,11 +130,28 @@ def get_ext_modules():
                                             "numba/_pymodule.h"],
                                    **np_compile_args)
 
-    ext_npyufunc_workqueue = Extension(
-        name='numba.npyufunc.workqueue',
-        sources=['numba/npyufunc/workqueue.c'],
-        depends=['numba/npyufunc/workqueue.h'])
-
+    tbb_root = os.getenv('TBBROOT')
+
+    if tbb_root:
+        print("Using TBBROOT=", tbb_root)
+        Q = '/Q' if sys.platform.startswith('win') or sys.platform == 'cygwin' else '-'
+        ext_npyufunc_workqueue = Extension(
+            name='numba.npyufunc.workqueue',
+            sources=['numba/npyufunc/tbbpool.cpp'],
+            depends=['numba/npyufunc/workqueue.h'],
+            include_dirs=[os.path.join(tbb_root, 'include')],
+            extra_compile_args=[Q+'std=c++11'],
```

Comment:
Yes, otherwise this will make building quite cumbersome. Many people don't use Intel C++, either for licensing or practical reasons.
(also, is Intel C++ binary-compatible with MSVC's CRT?)